### PR TITLE
Refactor remove search code from unit and lesson list LESQ-461

### PR DESCRIPTION
**Pull request by JBR90**
## Description

-Removes search related code from unit and lesson lists
-Move tracking events up to page level

Music year: 1958

## Issue(s)

Fixes #

## How to test

1. Go to https://deploy-preview-2126--oak-web-application.netlify.thenational.academy
Unit and lesson lists should look the same
Tracking events are being send when a item is clicked

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/__tests__/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessonListing.test.tsx [hunk 1]
```diff
@@ -93,4 +108,30 @@ describe("Lesson listing page", () => {
...
+  describe("tracking", () => {
+    test("It calls tracking.lessonSelected with correct props when clicked", async () => {
+      const { getByText } = render(
+        <LessonListPage curriculumData={lessonListingFixture()} />,
+      );
+
+      const lesson = getByText("Add two surds");
+
+      act(() => {
+        lesson.click();
```
Comment (njblore) [inline src/__tests__/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessonListing.test.tsx:120]: I'd recommend using `userEvent` to handle clicks wherever they're being tested to get as close to 'real' behaviour as possible - it also wraps in act so you shouldn't need to do it manually

Comment (JBR90) [reply src/__tests__/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessonListing.test.tsx:120]: Yeah I originally tried this but it wasn't calling the mocked tracking function, manually checked and avo was sending events. Maybe we could take a look at this together?

**Review by jack-skerrett-oak — APPROVED** [review]
Events still go through, components behave how I'd expect.

File: src/__tests__/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessonListing.test.tsx (hunk 1 above)
Comment (njblore) [reply src/__tests__/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessonListing.test.tsx:120]: oh good to know! that't a bit troubling though - were you `await` ing the userEvent.click?

Comment (JBR90) [reply src/__tests__/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessonListing.test.tsx]:
```
     act(async () => {
        await userEvent.click(lesson);
      });
```
I was using act - which failed the test
This worked  `  await userEvent.click(lesson);`

Comment (JBR90) [reply src/__tests__/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessonListing.test.tsx]:
```
    await act(async () => {
      await userEvent.click(unit);
    });
```
It was await! - was missing from the start of act on my previous attempt, this passes and removes act warning from tests.

**Review by njblore — APPROVED** [review]

**Pull request merged**